spec/javascripts/stimulus/unsaved_changes_controller_test.js

```diff
@@ -0,0 +1,195 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import { Application } from "stimulus"
+import unsaved_changes_controller from "../../../app/webpacker/controllers/unsaved_changes_controller"
+
+describe("UnsavedChangesController", () => {
+  beforeAll(() => {
+    const application = Application.start()
+    application.register("unsaved-changes", unsaved_changes_controller)
+  })
+
+  beforeEach(() => {
+    document.body.innerHTML = `
+      <form 
+        id="test-form" 
+        data-controller="unsaved-changes" data-action="beforeunload@window->unsaved-changes#leavingPage turbolinks:before-visit@window->unsaved-changes#leavingPage" 
+        data-unsaved-changes-changed="false"
+      >
+        <input id="test-checkbox" type="checkbox" data-action="change->unsaved-changes#formIsChanged"/>
+        <input id="test-submit" type="submit"/>
+      </form>
+    `
+  })
+
+  describe("#connect", () => {
+    describe("when disable-submit-button is true", () => {
```

Comment:
I don't see this condition used in the view. It's good to know it works, but perhaps it was unnecessary to specify this initial value if it's never used.

Reply:
> On step 4, 'save' and 'save and back to list' buttons are only active if data has been changed

This was the solution I used for the point above, before I decided against it after further testing. I kept it because I thought it might be useful if we were to use the controller in another context.